docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -9,57 +9,49 @@ in controller-runtime.
 It is suggested to use kubebuilder to initialize a project, and then you can
 follow the steps below to add admission webhooks for core types.
 
-## Implement Your Handler
+## Implement Your Webhook
 
-You need to have your handler implements the
-[admission.Handler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#Handler)
+You need to have your webhook to implement the
+[admission.CustomDefaulter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#CustomDefaulter)
 interface.
 
 ```go
-type podAnnotator struct {
-	Client  client.Client
-	decoder *admission.Decoder
-}
+type podAnnotator struct {}
 
-func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-	err := a.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
+func (a *podAnnotator) Default(ctx context.Context, obj runtime.Object) error {
+    var pod *corev1.Pod
 
-	// mutate the fields in pod
+    log := logf.FromContext(ctx)
 
-	marshaledPod, err := json.Marshal(pod)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
-}
-```
+    d, ok := obj.(*corev1.Pod)
+    if !ok {
+        return fmt.Errorf("expected a Pod but got a %T", obj)
+    }
 
-If you need a client, just pass in the client at struct construction time.
+	// mutate the fields in pod
 
-If you add the `InjectDecoder` method for your handler, a decoder will be
-injected for you.
+    log.Info("Annotated pod")
 
-```go
-func (a *podAnnotator) InjectDecoder(d *admission.Decoder) error {
-	a.decoder = d
 	return nil
 }
 ```
 
 **Note**: in order to have controller-gen generate the webhook configuration for
 you, you need to add markers. For example,
-`// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
+`// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
```

Comment:
```suggestion
`// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
```

I think this change is non-intentional, or?

Reply:
It's intentional: with 'core' Group, the group name is empty so the `/mutate-<group>-<version>-<kind>` path becomes `/mutate--v1-pod`. See https://github.com/kubernetes-sigs/controller-runtime/pull/2314

Reply:
I see. I would prefer a note/comment around this in the docs (here). It might not just be me that finds this strange and assume it's non-intentional.

Reply:
@erikgb thanks, I added a note in parenthesis about the expected format.